packages/imperative/src/plugins/AbstractPluginLifeCycle.ts

```diff
@@ -0,0 +1,61 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+/**
+ * An optional module of a plugin extends this class to
+ * perform actions during the installation lifecycle of a plugin.
+ * A plugin identifies that it supplies a life cycle class
+ * by specifying the 'pluginLifeCycle' imperative configuration property.
+ */
+export abstract class AbstractPluginLifeCycle {
```

Comment:
Since this is an abstract class, I believe all of its methods will be required which means that if a plugin wants to add a "postInstall" lifecycle script, they will also have to add a "preUninstall" lifecycle script even if they don't need one and leave it empty.

If we add more lifecycle events in the future, we may want to convert the class to an interface where each lifecycle hook is optional so that plugin devs don't have to define them all. This seems fine for now - I'm ok with requiring both since making them optional in the future wouldn't be a breaking change 🙂

Reply:
My first attempt at implementation was with an interface. It seemed to fit the model better. I had great difficulty (as in I could not get it to work) with the existing PluginManagementFacility's plugin module requirer callback function.

Perhaps I was just confused, but things did not start to work until I switched to loading a class that extended an abstract class.